### PR TITLE
Adding event macros for event stream consumption

### PR DESCRIPTION
**Pull request by jeffgrunewald**
co-authored-by: Nathaniel Stevens <nstevens@pillartechnology.com>

### CODE REVIEW

File: lib/smart_city/events.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  Defines macros for encoding event types the Smart City platform
+  will respond to in any of the various micro service components
+  in a central location shared by all components.
+  """
+
+  @doc """
+  Defines an update event to a dataset within the system. The
+  system treats create events as a subset of updates.
+  """
+  defmacro update_dataset(), do: "dataset:update"
```
Comment (ManApart) [inline lib/smart_city/events.ex]: Why do these need to be macros as opposed to just functions?

Comment (bbalser) [reply lib/smart_city/events.ex]: So they can be used to pattern match in handle_event functions.

Comment (jeffgrunewald) [reply lib/smart_city/events.ex]: Yeah, you can’t use functions as a pattern matcher in a function head so the macro just evaluates to that string at compile time.

Comment (ManApart) [reply lib/smart_city/events.ex]: Ah! That makes sense.

**Pull request merged**